static/fetch.js

```diff
@@ -0,0 +1,50 @@
+(function(){
+	const RELOAD_TIME = 5000;
+	setInterval(loop, RELOAD_TIME);
+})();
+
+async function loop(){
+	let data = await reload();
+	for (let printer_name in data){
+		queue(printer_name, data[printer_name]);
+	}
+}
+
+function element(username, date){
+	let wrapper = document.createElement('div');
+	wrapper.classList.add('printlist-user');
+	let name = document.createElement('div');
+	name.classList.add('printlist-user-handle');
+	name.innerText = username.replace(/\</g, '&lt;').replace(/\>/g, '&gt;');
+	wrapper.appendChild(name);
+	setTimeout(()=>wrapper.remove(), 180000)
+	return wrapper;
+}
+
+function queue(printer_name, data){
+	data.sort((a, b)=>{
+		// Sorts by date
+		return a[1]-b[1];
+	})
+	.map(item=>([item[0], new Date(item[1]* 1000)]))
+	.forEach(item=>{
+		let parent = document.querySelector(`.${printer_name} h1`);
+		if (document.querySelector(`.${printer_name} .printlist-null`)) document.querySelector(`.${printer_name} .printlist-null`).remove();
+		parent.after(element(item[0], item[1]));
+		// Do whatever you need, [username, date as Date Object]
+	});
+}
+
+function reload(){
+	return new Promise(async (resolve, reject)=>{		
```

Comment:
It would probably be a lot cleaner to use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) in this function instead of `XMLHttpRequest`